### PR TITLE
Benchmark Slots and Auction

### PR DESCRIPTION
**Pull request by shawntabrizi**
in progress

### CODE REVIEW

File: runtime/common/src/slots.rs [hunk 1]
```diff
@@ -1751,3 +1751,48 @@ mod tests {
 		});
 	}
 }
+
+#[cfg(feature = "runtime-benchmarks")]
```
Comment (xlc) [inline runtime/common/src/slots.rs:1755]: this don't deserve a new file? this file is already > 1500 lines...

**Comment by coriolinus** [comment]
Related to #1512 also?

File: runtime/common/src/slots.rs [hunk 2]
```diff
@@ -530,8 +532,10 @@ decl_module! {
 					// Should have already begun. Remove the on-boarding entry and register the
 					// parachain for its immediate start.
 					<Onboarding<T>>::remove(&para_id);
-					let _ = T::Parachains::
+					// TODO: Probably best to make sure this doesn't fail or handle fail conditions.
```
Comment (gavofyork) [inline runtime/common/src/slots.rs:535]: stray TODO

**Comment by shawntabrizi** [comment]
replaced elsewhere

**Pull request closed without merging**